app/src/main/java/com/lambdaschool/hackathon_portal/ui/fragments/details/DetailsFragment.kt

```diff
@@ -0,0 +1,154 @@
+            }
+        }
+
+        if (hackathonId != null) {
+            detailsViewModel.getHackathon(hackathonId).observe(this, Observer {
+                if (it != null) {
+                    updateHackathonViews(it)
+                } else {
+                    activity?.apply {
+                        Toast.makeText(this,
+                            "Failed to get Hackathon",
+                            Toast.LENGTH_LONG).show()
+                    }
+                }
+            })
+        }
+
+        fragment_details_fab_save_hackathon.setOnClickListener {
+            val newHackathon =
+                Hackathon(fragment_details_edit_text_hackathon_name.text.toString(),
+                    fragment_details_edit_text_hackathon_description.text.toString(),
+                    fragment_details_edit_text_hackathon_url.text.toString(),
+                    fragment_details_edit_text_hackathon_start_date.text.toString(),
+                    fragment_details_edit_text_hackathon_end_date.text.toString(),
+                    fragment_details_edit_text_hackathon_location.text.toString(),
+                    switchState)
+
+            if (hackathonId != null) {
+                detailsViewModel.updateHackathon(hackathonId, newHackathon)
+                    .observe(this, Observer {
+                    if (it != null) {
+                        updateHackathonViews(it)
+                        activity?.apply {
+                            Toast.makeText(this,
+                                "Successfully updated Hackathon",
+                                Toast.LENGTH_LONG).show()
+                        }
+                    } else {
+                        activity?.apply {
+                            Toast.makeText(this,
+                                "Failed to update Hackathon",
+                                Toast.LENGTH_LONG).show()
+                        }
+                    }
+                })
+            }
+        }
+
+        fragment_details_fab_delete_hackathon.setOnClickListener {
+            val title = "Delete Hackathon?"
+            val msg = "Are you sure you want to delete this Hackathon?"
+
+            AlertDialog.Builder(context!!)
+                .setTitle(title)
+                .setMessage(msg)
+                .setPositiveButton("Yes") { _, _ ->
+                    if (hackathonId != null) {
+                        detailsViewModel.deleteHackathon(hackathonId)
+                            .observe(this, Observer {
+                                if (it != null) {
+                                    if (it) {
+                                        activity?.apply {
+                                            Toast.makeText(this,
+                                                "Successfully deleted Hackathon",
+                                                Toast.LENGTH_LONG).show()
+                                            findNavController().navigateUp()
```

Comment:
Injecting NavController would be better performance wise.

adding to line 34
```
@Inject
lateinit var navController: NavController
```
```suggestion
                                            navController.navigateUp()
```

Reply:
Changed in commit 770119ff532eda3f5b007d568d5ecbf612792a90